### DISCUSSION TITLE
Unlock protocol and next.id avatar DID.

### DISCUSSION THREAD

**Discussion by javaspeak** [NextDotID/proof_server#107] [Q&A]
With the unlock protocol one can create something similar an attestation that someone is for example a "good software developer".  

Was wondering if there was a way to link that accomplishment to the next.id avatar?

This would be quite neat for third parties could create this attestation and the DID owner could decide whether they want it added to their avatar or not.

Need to research unlock protocol more - just thought this would be very neat if possible.

Maybe someone who knows more about unlock protocol can answer me.

**Comment by javaspeak** [comment]
<img width="767" alt="Screenshot 2024-01-30 at 12 36 46" src="https://github.com/NextDotID/proof_server/assets/26520180/22d02b00-8ca6-4f4f-8103-6ddbd6f9bf5f">

Here have integrated UTU Trust with next.id.  People are giving signal on a DID.

What I want to be able to do is to attach a Unlock accediation to a next.id Avatar DID.

@nykma do you think this can be possible?

This explains the unlock certification:

https://docs.unlock-protocol.com/getting-started/what-is-unlock/litepaper#certifications-and-credentials

They can be issued for free (only gas charges apply)

I think it will kick ass if we can link them to next.id avatar DIDs

**Comment by javaspeak** [comment]
This Contract Address:

     https://sepolia.etherscan.io/address/0x1ADd1cb51D744E3006ACa2cdd886C86A2fA391E3


Shows that the address:

      0x1ADd1cb51D744E3006ACa2cdd886C86A2fA391E3

was granted this Unlock NFT Certification:   

     [Great Vegan Chef (KEY)]

See:  https://sepolia.etherscan.io/token/0x1add1cb51d744e3006aca2cdd886c86a2fa391e3?a=0x8dd1cc73d574608ba7e27c485da4e3645c62fbf6

**Comment by javaspeak** [comment]
<img width="707" alt="Screenshot 2024-02-02 at 18 17 08" src="https://github.com/NextDotID/proof_server/assets/26520180/14d67d7d-6cc9-4c21-89a6-d395881ab40a">

When you connect the wallet to the url:

https://app.unlock-protocol.com/keychain

It shows what Unlock NFT Certifications one has

**Comment by javaspeak** [comment]
<img width="1038" alt="Screenshot 2024-02-02 at 18 20 22" src="https://github.com/NextDotID/proof_server/assets/26520180/a1b15c6a-5fa4-4234-836f-8790e14ede15">

This links also shows the certification but only if you connect with the wallet that was give the certification

**Comment by javaspeak** [comment]
So we can know that an ethereum address has the NFT Certification by verifying here:

https://sepolia.etherscan.io/token/0x1add1cb51d744e3006aca2cdd886c86a2fa391e3?a=0x8dd1cc73d574608ba7e27c485da4e3645c62fbf6

@nykma how could we add the NFT to the next.id avatar?

In the above link we can add the wallet address 0x1add1cb51d744e3006aca2cdd886c86a2fa391e3? to the avatar. 

However not sure how to add the Unlock NFT certification to the next.id avatar.  Any ideas?

Am a bit of a newbee to all of this and would LOVE if one can decide which unlock certifications one can add to one's avatar.

**Comment by nykma** [comment]
> With the unlock protocol one can create something similar an attestation that someone is for example a "good software developer".
> Was wondering if there was a way to link that accomplishment to the next.id avatar?

Technically this "badge" is just a type of NFT, which is not a dedicate identity (because ERC721 must belong to a specific wallet)

But, we're now working on a "GitCoin Passport"-ish product to help user quickly build reputation in cyberworld

**Comment by javaspeak** [comment]
> > With the unlock protocol one can create something similar an attestation that someone is for example a "good software developer".
> > Was wondering if there was a way to link that accomplishment to the next.id avatar?
> 
> Technically this "badge" is just a type of NFT, which is not a dedicate identity (because ERC721 must belong to a specific wallet)
> 
> But, we're now working on a "GitCoin Passport"-ish product to help user quickly build reputation in cyberworld

When creating the Unlock membership there is an option to make it NOT transferable which means it is connected to a specific wallet.

Unlock also allows them to expire.

**Reply by nykma** [reply]
Since this is a discussion post under `proof_server` repo,  I would say that the unlock membership acts more like a "authorization" level stuff (or you can call it a "privilege"), while `proof_service` focus more on "authentication" level (which means an independent identity that doesn't need to belongs to other stuff)

Again, this does not mean that we NextID has no interest in integrating "label", "privilege" or "trait" of an identity into the NextID ecosystem. We will reach it in other way, but not in ProofService for now.

**Comment by javaspeak** [comment]
> > With the unlock protocol one can create something similar an attestation that someone is for example a "good software developer".
> > Was wondering if there was a way to link that accomplishment to the next.id avatar?
> 
> Technically this "badge" is just a type of NFT, which is not a dedicate identity (because ERC721 must belong to a specific wallet)
> 
> But, we're now working on a "GitCoin Passport"-ish product to help user quickly build reputation in cyberworld

Will this "GitCoin Passport" --ish product be using he same next.id avatars?

And is the product for managing your avatar or is it about viewing the avatar or both?

What I want is that on a social media platform is that one can view an avatar and all the handles associated with it and also see certifications.

So the idea with a certification is someone can give it to someone and then that person can decide if they want it added to their avatar or not.  I thought that maybe Unlock would be a good fit with next.id avatars.

Can you advise what your would recommend me doing @nykma ?

**Reply by nykma** [reply]
> What I want is that on a social media platform is that one can view an avatar and all the handles associated with it and also see certifications.

Check https://web3.bio to see if it meets your demand. It is a frontend of `relation_server`

**Reply by javaspeak** [reply]
The way I understand it, is that the relation_server is like a search engine which finds stuff connected to a wallet address.

So it may or may not find a Unlock membership associated with an address via an unlock smart contract, token id and wallet address.

Was thinking that one could maybe have a new platform type in the proof server which is an IPNS domain hash.  This IPNS hash can be mapped by the owner of the IPNS hash to an IPFS content hash.

So it is possible that the IPNS hash points to an IPFS hash which has some json it it which lists etherscan urls which prove the user has certain Unlock memberships or credentials.

If the unlock memberships are configured as non transferable then this new platform type called IPNS could support what I want to do.

See here @nykma - this is like brain storming - there could be better ways - was just thinking of a generic way of allowing a user to control facets of their identity to the outside world.
